### ISSUE TITLE
Add Skip to Main Content link

### ISSUE THREAD

**Issue by atsansone** [flutter/website#10531]
From https://buganizer.corp.google.com/issues/337077761

### Observed Results

Navigating through the page, noticed that the skip to main content link is missing.

* [Screencast VoiceOver](http://screencast/cast/NTEwNDkxNzA3MjA1MjIyNHxhNDViNjU2NC1iNg)
* [Screencast ChromeVox](http://screencast/cast/NTM2NjIyMjg4MjM0MDg2NHw2MDY0MWM3MC02OA)

### Expected Results

Ensure that when the  user is navigating in the page the skip to main content link should be displayed and work accordingly for going to the main content.

### User Impact

Screen reader and Cognitive users have issues because they cannot bypass repeated headings and blocks of information on the page, they need to navigate everything before reaching the  main content.

### Steps to Reproduce

1. Enabled VoiceOver using <kbd>Command</kbd> + <kbd>F5</kbd> also ChromeVox using <kbd>Ctrl</kbd> + <kbd>Alt</kbd> + <kbd>Z</kbd>.
1. Open the following page: https://docs.flutter.dev/ui/widgets
1. Start the linear navigation using <kbd>Tab</kbd>.
1. Observe that the skip to main content link is not displayed on the page.

### GAR Assessment

[Flutter Docs](https://a11ytracker.corp.google.com/products/c2fdcf18-53d2-475d-b8b0-e5dedf8cb900?e=ProductLabels)

### GAR Details

[2.8. Do you provide a bypass for headings and content that are repeated across multiple pages or within the same page?](go/GAR-web#repeated-content)

### Environment Details

Machine: MAC/ChromeBook
Browser: Chrome Version 120.0.6099.272 (Official Build) (64-bit)
Screen reader: VoiceOver/ChromeVox.
Bug filling time: 15 mins.
